coordinator/src/orderbook/collaborative_revert.rs

```diff
@@ -0,0 +1,83 @@
+use crate::db::channels;
+use crate::db::collaborative_reverts;
+use crate::message::NewUserMessage;
+use crate::message::OrderbookMessage;
+use crate::position::models::parse_channel_id;
+use anyhow::bail;
+use anyhow::Result;
+use bitcoin::secp256k1::PublicKey;
+use diesel::r2d2::ConnectionManager;
+use diesel::r2d2::Pool;
+use diesel::PgConnection;
+use futures::future::RemoteHandle;
+use futures::FutureExt;
+use orderbook_commons::Message;
+use tokio::sync::broadcast;
+use tokio::sync::mpsc;
+
+pub fn monitor(
+    pool: Pool<ConnectionManager<PgConnection>>,
+    tx_user_feed: broadcast::Sender<NewUserMessage>,
+    notifier: mpsc::Sender<OrderbookMessage>,
+) -> RemoteHandle<Result<()>> {
+    let mut user_feed = tx_user_feed.subscribe();
+    let (fut, remote_handle) = async move {
+        while let Ok(new_user_msg) = user_feed.recv().await {
+            tokio::spawn({
+                let mut conn = pool.get()?;
+                let notifier = notifier.clone();
+                async move {
+                    tracing::debug!(trader_id=%new_user_msg.new_user, "Checking if the user needs to be notified about collaboratively reverting a channel");
+                    if let Err(e) = process_pending_collaborative_revert(&mut conn, notifier, new_user_msg.new_user).await {
+                        tracing::error!("Failed to process pending collaborative revert. Error: {e:#}");
+                    }
+                }
+            });
+        }
+        Ok(())
+    }.remote_handle();
+
+    tokio::spawn(fut);
+
+    remote_handle
+}
+
+/// Checks if there are any pending collaborative reverts
+async fn process_pending_collaborative_revert(
+    conn: &mut PgConnection,
+    notifier: mpsc::Sender<OrderbookMessage>,
+    trader_id: PublicKey,
+) -> Result<()> {
+    if let Some(channel) = channels::get_by_trader_pubkey(trader_id, conn)? {
```

Comment:
🔧 I guess this will produce ugly error messages on the coordinator, as this will be run for every user opening the app. They might have an announced channel that never got funded, that would produce "Can't revert a channel ..." error. Even though we didn't even intended to do that.

I propose you turn that logic around. Firstly fetch per `trader_id` if there is a pending `collaborative_revert` then check if the `channel_id` is still correct. Not sure if the last check is actually required though.

Reply:
Good point. 

done: [`fe4c71f` (#1478)](https://github.com/get10101/10101/pull/1478/commits/fe4c71fceb1fb056cc90104ac96acfc83c4994ce)